test/with_dummyserver/test_socketlevel.py

```diff
@@ -966,6 +967,63 @@ def socket_handler(listener: socket.socket) -> None:
             assert pool.pool.qsize() == 1
             assert response.connection is None
 
+    def test_socket_close_socket_then_file(self) -> None:
+        def consume_ssl_socket(listener: socket.socket) -> None:
+            try:
+                with listener.accept()[0] as sock, original_ssl_wrap_socket(
+                    sock,
+                    server_side=True,
+                    keyfile=DEFAULT_CERTS["keyfile"],
+                    certfile=DEFAULT_CERTS["certfile"],
+                    ca_certs=DEFAULT_CA,
+                ) as ssl_sock:
+                    consume_socket(ssl_sock)
+            except (ConnectionResetError, ConnectionAbortedError, OSError):
+                pass
+
+        self._start_server(consume_ssl_socket)
+        with socket.create_connection(
+            (self.host, self.port)
+        ) as sock, contextlib.closing(
+            ssl_wrap_socket(sock, server_hostname=self.host, ca_certs=DEFAULT_CA)
+        ) as ssl_sock, ssl_sock.makefile(
+            "rb"
+        ) as f:
+            ssl_sock.close()
+            f.close()
+            # SecureTransport is supposed to raise OSError but raises
+            # ssl.SSLError when closed because ssl_sock.context is None
+            with pytest.raises((OSError, ssl.SSLError)):
+                ssl_sock.sendall(b"hello")
+            assert ssl_sock.fileno() == -1
+
+    def test_socket_close_stays_open_with_makefile_open(self) -> None:
+        def consume_ssl_socket(listener: socket.socket) -> None:
+            try:
+                with listener.accept()[0] as sock, original_ssl_wrap_socket(
+                    sock,
+                    server_side=True,
+                    keyfile=DEFAULT_CERTS["keyfile"],
+                    certfile=DEFAULT_CERTS["certfile"],
+                    ca_certs=DEFAULT_CA,
+                ) as ssl_sock:
+                    consume_socket(ssl_sock)
+            except (ConnectionResetError, ConnectionAbortedError, OSError):
+                pass
+
+        self._start_server(consume_ssl_socket)
+        with socket.create_connection(
+            (self.host, self.port)
+        ) as sock, contextlib.closing(
+            ssl_wrap_socket(sock, server_hostname=self.host, ca_certs=DEFAULT_CA)
+        ) as ssl_sock, ssl_sock.makefile(
+            "rb"
+        ):
+            ssl_sock.close()
```

Comment:
Why are we calling `close()` twice here, for coverage? Can we pull this into a separate test if so?

Reply:
It's a regression test, the previous code closed the socket when .close() is called twice even if a makefile is open

Reply:
> Can we pull this into a separate test if so?

I'm not sure what you mean by this

Reply:
Was suggesting moving the double close to a separate test if it was only for coverage, since this is required to cause the bug we should keep it as-is, thanks for the explanation.